src/components/Form/Form.js

```diff
@@ -50,100 +50,95 @@ export function Form({ children, config }) {
     // If we wanted to have a form be optional, you could have its initial
     // value set to empty string, so the form could be submitted without
     // the field being touched.
-    initialValues[x] = 'No answer supplied for ' + x
+    initialValues[x] = `Initial invalid state for ${x}`
   })
 
   // Hooks
   const [
-    getInputErrors,
-    getInputValues,
+    getFieldErrors,
+    getFieldValues,
     setStateFactory,
     getFormErrorMessage,
     setFormErrorMessage,
     getFormIsValid,
   ] = useFormState(initialValues)
 
   // Wrapper for default <form> submit function.
-  async function onSubmit(syntheticReactEvent) {
-    // Set the form-level error to disappear while the form tries to submit.
+  async function onSubmit(ev) {
     setFormErrorMessage('')
-
-    // Prevents normal form submission redirection
-    syntheticReactEvent.preventDefault()
+    ev.preventDefault()
 
     // Check hook state to see if the form has any errors registered.
     // Ideally, forms should not have the submit button be clickable unless
     // all errors are clear by passing down getFormIsValid() to the `disabled`
     // prop on the submit button, but this is a backup in case they don't.
     if (!getFormIsValid()) {
-      if (getInputErrors()) {
-        setFormErrorMessage(
-          'One of the inputs was invalid. Errors: ' + getInputErrors()
-        )
+      if (getFieldErrors()) {
```

Comment:
Nit: this doesn't need to be nested since it looks like if `getFormIsValid` is `truthy`, we `return`. e.g.

```js
    if (getFormIsValid) {
        return
    }

    if (getFieldErrors()) {
        setFormErrorMessage('Errors: ' + getFieldErrors())
    } else {
        setFormErrorMessage("You haven't typed anything yet")
    }
```